src/Core/src/Signer/SignerV4.php

```diff
@@ -245,6 +253,13 @@ private function convertBodyToStream(Request $request, \DateTimeInterface $now,
             $contentLength = $body->length();
         }
 
+        // If content length is unknown, use the rewindable stream to read it once locally in order to get the length
+        if (null === $contentLength) {
+            $request->setBody($body = RewindableStream::create($body));
+            $body->travers();
```

Comment:
This function could be private, right?

Reply:
not sure how. 
For instance here I want to get the exact value of length. While in AbstractApi, it's better to get length, but if wwe don't, it's not compulsotry.

Reply:
In the `RewindableStream::length()`:

```php
    public function length(): ?int
    {
        if (null !== $this->fallback) {
            $this->read():
        }

         return $this->fallback->length();
    }
```

Is this a good idea? Will it work for `stringify()` and `hash()` too?


Hm... maybe no. Because that will always duplicate the stream. Sometimes there is no need to.

Reply:
yes, that were my point but ou explained it better than me :)